### DISCUSSION TITLE
Spell 관련 내용

### DISCUSSION THREAD

**Discussion by giftof** [42StudyGame/RandomDiceDefense42-Client#64] [General]
Task에 버프라고 써놓고, dot를 만들었던것 같습니다.
버프의 범위를 다시 산정하고 코드 수정할 계획입니다.

Spell에는, buff, aura, dd, dot이렇게 4가지로 나누면 될것 같구요

Buff: 일정 시간동안 유효한 효과
Aura: 일정 범위동안 유효한 효과
Dd: 단발성 효과 (durect damage)
Dot: 일정시간동안 일정 간격마다 유효한 효과 (damage of time) (dd에 반복 주기와 횟수를 추가로 가진다고 봐도 될것 같습니다)

용어에는 damage가 들어가지만, 피해량이 아닐수도 있습니다. (이로운 효과일수도..)

제가 잡았던 Buff라고 되어있던 task는, 저중에서 dot를 한것 같으므로.. buff를 다시 진행하도록 하겠습니다.

그리고, spell범주를 위처럼 나누는것에 대한 의견들도 부탁드립니다.

그리고, 그리고, spell과 skill을 구분해서 생각할지, 퉁쳐서 생각할지도 의견들 주세요~

**Comment by nonoaa** [comment]
spell과 skill은 다른점이 무엇일까요?

**Reply by giftof** [reply]
제가 아는 Rpg게임에서는..
Spell은 mana를 사용하고
Skill은 stamina를 사용합니다

이건, 하이브리드 클래스를 위해서(?) 구분하는 경우가 있긴 합니다.

**Reply by nonoaa** [reply]
각각 사용하는 자원이 다르다고 보면 될까요? RPG를 많이 안해본 저에게는 생소한 개념이라 잘 와닿지는 않네요

**Reply by yugd1211** [reply]
유저가 직접 적용하는 스킬이 머지하거나 타워 위치 바꾸는 이정도 밖에 없을거 같아서 스킬은 굳이 나눌 필요는 없을거 같습니다.

**Reply by giftof** [reply]
네 사용하는 자원이 다르다고 보시면 될듯요. (실제로도 이정도입니다만, 이로인해 파생되는 사항이 다르다고 알고있어요)
